### DISCUSSION TITLE
What directory should be persisted if running in a stateless environment?

### DISCUSSION THREAD

**Discussion by 2fst4u** [LukeChannings/moviematch#86] [General]
I realise most of the functionality doesn't really care if the application starts from scratch after a reboot, but I run the app in kubernetes where the pod it's running in can be killed at any moment for various reasons by design.

So to that end, is there a directory that can be persisted to make sure rooms and votes stay live after a full restart of the container? 

This would also be good for running multiple identical pods for load balancing (because hell, why not?)

**Comment by LukeChannings** [comment]
Rooms are currently ephemeral in both 1.0 and 2.0, and will be reset if the server is restarted no matter what.
I was thinking about adding basic persistence for 2.0, since Deno recently added support for Web Storage APIs.

MovieMatch's model isn't built against an ORM, with the state being in-memory - so adding scalable persistence is out of reach without a lot of work. I wouldn't worry about performance (and load balancing) until it actually becomes a problem, but I don't think it will be a problem until you have 10s of thousands of simultaneous users.

I'm also [interested in kubernetes](https://github.com/LukeChannings/kube-config) (and I run MovieMatch in my cluster), but MovieMatch isn't really architected to support multiple instances.

You could use sticky sessions to make sure a user never hits an instance that doesn't have a room that they created, and set up and a caching proxy (e.g. Varnish) to speed up the Plex API calls, but it wouldn't solve the hard problem of a new user coming in with a share link - how do they get directed to the correct instance?
